core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -1103,14 +1136,18 @@ class ReplicaFetcherThreadTest {
     assertProcessPartitionDataWhen(isReassigning = false)
   }
 
-  @Test
-  def testBuildFetch(): Unit = {
+  @ParameterizedTest
+  @ValueSource(booleans = Array(true, false))
+  def testBuildFetch(usingReplicaState: Boolean): Unit = {
```

Comment:
Instead of hijacking this test, could we add a new and simpler unit test focused on what we want to test? Note that you could use `@EnumSource(value = MetadataVersion.class, names = {"IBP_3_0_IV1", "IBP_3_3_IV2"})` (with the relevant IBPs).

Reply:
Actually, for the ReplicaFetchThread, we don't have anything to test. We have tested RemoteLeaderEndPoint.